driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -883,6 +899,65 @@ public String toString() {
         }
     }
 
+    private static class OutStage implements Bson {
+
+        private final String collectionName;
+        private final AggregateOutStageOptions options;
+
+        OutStage(final String collectionName, final AggregateOutStageOptions options) {
+            this.collectionName = collectionName;
+            this.options = options;
+        }
+
+        @Override
+        public <TDocument> BsonDocument toBsonDocument(final Class<TDocument> tDocumentClass, final CodecRegistry codecRegistry) {
+            BsonDocumentWriter writer = new BsonDocumentWriter(new BsonDocument());
+            writer.writeStartDocument();
+
+            writer.writeName("$out");
+            if (optionsAreAllDefault()) {
```

Comment:
This conditional is important.  It means that even if you use the new out method, so long as all the options are set to the default values then it uses the pre-4.2 syntax.  Only if you opt-in to 4.2 features does it use the 4.2 syntax.